packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -143,15 +147,36 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
         const newNavigatedDate = navigateToNewDate(gridNavigatedDate, 'Day', -1, restrictedDatesOptions);
 
         if (!!newNavigatedDate) {
+          setShouldFocusInDayGrid(true);
           setGridNavigatedDate(newNavigatedDate);
         }
       },
     },
     rtl: context.rtl,
   });
 
-  // TODO: make new date instances, so they are less mutable
-  const [gridNavigatedDate, setGridNavigatedDate] = React.useState<Date>(navigatedDate || today || new Date());
+  const [gridNavigatedDate, setGridNavigatedDate] = React.useState<Date>(
+    () => new Date((navigatedDate || today || new Date()).getTime()),
+  );
+
+  const normalizeDateInGrid = (date: Date): Date => {
+    const result = new Date(date.getTime());
+    result.setDate(1);
+    return result;
+  };
+
+  const getRidOfSecondsMinutesHours = (date: Date): Date => {
+    const result = new Date(date.getTime());
```

Comment:
```suggestion
    const result = new Date(date.getFullYear(), date.getMonth(), date.getMinutes());
```
Like in https://github.com/microsoft/fluentui/blob/5fbb948524f5c80cf59378e57ac898fabd6f3273/packages/date-time-utilities/src/dateGrid/getDayGrid.ts#L37?